### ISSUE TITLE
[youtube] Format `18` no longer available

### ISSUE THREAD

**Issue by anikialt** [yt-dlp/yt-dlp#11154]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm asking a question and **not** reporting a bug or requesting a feature
- [X] I've looked through the [README](https://github.com/yt-dlp/yt-dlp#readme)
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar questions **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Please make sure the question is worded well enough to be understood

audio track download separately from video

Lastest version of dlp

### Provide verbose output that clearly demonstrates the problem

- [ ] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [ ] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

**Comment by seproDev** [comment]
You will need to provide a bit more info. What exactly is your problem? The verbose log is also not optional.

**Comment by pdkl95** [comment]
I suspect I've been seeing the same problem: over the last ~12-24 hours, format 18 started disappearing from some videos at first, and is now gone from any video i've tried to download in the last couple hours. Maybe format 18 has been dropped? I've always prefered the combined formats, but that may no longer be available... /sigh/

(this is using the default "ios"+"web creator" players. I haven't tried any other player)

if it helps, a log showing the lack of format 18:

```
yt-dlp \
    -f 'b' \
    --replace-in-metadata 'uploader_id' '^[@]' '' \
    --console-title \
    --continue \
    --restrict-filenames \
    --output '%(uploader_id)s-%(title)s-%(upload_date)s.%(ext)s' \
    --force-ipv4 \
    --sleep-interval '3' \
    --max-sleep-interval '6' \
    --fixup 'warn' \
    --no-colors \
    --abort-on-error \
  --verbose \
  -F \
  https://www.youtube.com/watch?v=UfqBN6w06qI
[debug] Command-line config: ['-f', 'b', '--replace-in-metadata', 'uploader_id', '^[@]', '', '--console-title', '--continue', '--restrict-filenames', '--output', '%(uploader_id)s-%(title)s-%(upload_date)s.%(ext)s', '--force-ipv4', '--sleep-interval', '3', '--max-sleep-interval', '6', '--fixup', 'warn', '--no-colors', '--abort-on-error', '--verbose', '-F', 'https://www.youtube.com/watch?v=UfqBN6w06qI']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version nightly@2024.10.01.232843 from yt-dlp/yt-dlp-nightly-builds [e59c82a74] (pip)
[debug] Python 3.10.5 (CPython x86_64 64bit) - Linux-3.18.11-fixed-x86_64-Intel-R-_Core-TM-2_Quad_CPU_Q9650_@_3.00GHz-with-glibc2.23 (OpenSSL 1.1.1q  5 Jul 2022, glibc 2.23)
[debug] exe versions: ffmpeg 4.3.1-static, ffprobe 4.3.1-static, phantomjs 2.1.1, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.20.0, brotli-1.0.9, certifi-2024.08.30, curl_cffi-0.5.10, mutagen-1.46.0, requests-2.32.3, sqlite3-3.21.0, urllib3-2.2.2, websockets-13.1
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets, curl_cffi
[debug] Loaded 1838 extractors
[youtube] Extracting URL: https://www.youtube.com/watch?v=UfqBN6w06qI
[youtube] UfqBN6w06qI: Downloading webpage
[youtube] UfqBN6w06qI: Downloading ios player API JSON
[youtube] UfqBN6w06qI: Downloading web creator player API JSON
[youtube] UfqBN6w06qI: Downloading m3u8 information
[debug] Sort order given by extractor: quality, res, fps, hdr:12, source, vcodec:vp9.2, channels, acodec, lang, proto
[debug] Formats sorted by: hasvid, ie_pref, quality, res, fps, hdr:12(7), source, vcodec:vp9.2(10), channels, acodec, lang, proto, size, br, asr, vext, aext, hasaud, id
[debug] Replacing all '^[@]' in uploader_id with ''
[MetadataParser] Changed uploader_id to: tfolsenuclear
[info] Available formats for UfqBN6w06qI:
ID      EXT   RESOLUTION FPS CH |   FILESIZE   TBR PROTO | VCODEC          VBR ACODEC      ABR ASR MORE INFO
-----------------------------------------------------------------------------------------------------------------------------
sb3     mhtml 48x27        0    |                  mhtml | images                                  storyboard
sb2     mhtml 80x45        0    |                  mhtml | images                                  storyboard
sb1     mhtml 160x90       0    |                  mhtml | images                                  storyboard
sb0     mhtml 320x180      0    |                  mhtml | images                                  storyboard
233     mp4   audio only        |                  m3u8  | audio only          unknown             Default, IOS
234     mp4   audio only        |                  m3u8  | audio only          unknown             Default, IOS
139-drc m4a   audio only      2 |    7.19MiB   49k https | audio only          mp4a.40.5   49k 22k low, DRC, IOS, m4a_dash
139     m4a   audio only      2 |    7.19MiB   49k https | audio only          mp4a.40.5   49k 22k low, IOS, m4a_dash
140-drc m4a   audio only      2 |   19.08MiB  129k https | audio only          mp4a.40.2  129k 44k medium, DRC, IOS, m4a_dash
140     m4a   audio only      2 |   19.08MiB  129k https | audio only          mp4a.40.2  129k 44k medium, IOS, m4a_dash
602     mp4   256x144     15    | ~ 15.53MiB  105k m3u8  | vp09.00.10.08  105k video only          IOS
269     mp4   256x144     30    | ~ 29.64MiB  201k m3u8  | avc1.4D400C    201k video only          IOS
160     mp4   256x144     30    |   11.31MiB   77k https | avc1.4D400C     77k video only          144p, IOS, mp4_dash
603     mp4   256x144     30    | ~ 24.77MiB  168k m3u8  | vp09.00.11.08  168k video only          IOS
278     webm  256x144     30    |    9.93MiB   67k https | vp09.00.11.08   67k video only          144p, IOS, webm_dash
229     mp4   426x240     30    | ~ 55.71MiB  378k m3u8  | avc1.4D4015    378k video only          IOS
133     mp4   426x240     30    |   24.96MiB  169k https | avc1.4D4015    169k video only          240p, IOS, mp4_dash
604     mp4   426x240     30    | ~ 42.07MiB  286k m3u8  | vp09.00.20.08  286k video only          IOS
242     webm  426x240     30    |   18.90MiB  128k https | vp09.00.20.08  128k video only          240p, IOS, webm_dash
230     mp4   640x360     30    | ~107.99MiB  733k m3u8  | avc1.4D401E    733k video only          IOS
134     mp4   640x360     30    |   46.54MiB  316k https | avc1.4D401E    316k video only          360p, IOS, mp4_dash
605     mp4   640x360     30    | ~ 77.52MiB  526k m3u8  | vp09.00.21.08  526k video only          IOS
243     webm  640x360     30    |   34.83MiB  236k https | vp09.00.21.08  236k video only          360p, IOS, webm_dash
231     mp4   854x480     30    | ~182.31MiB 1237k m3u8  | avc1.4D401F   1237k video only          IOS
135     mp4   854x480     30    |   88.96MiB  604k https | avc1.4D401F    604k video only          480p, IOS, mp4_dash
606     mp4   854x480     30    | ~124.23MiB  843k m3u8  | vp09.00.30.08  843k video only          IOS
244     webm  854x480     30    |   59.33MiB  403k https | vp09.00.30.08  403k video only          480p, IOS, webm_dash
311     mp4   1280x720    60    | ~410.66MiB 2787k m3u8  | avc1.640020   2787k video only          IOS
298     mp4   1280x720    60    |  260.30MiB 1767k https | avc1.640020   1767k video only          720p60, IOS, mp4_dash
612     mp4   1280x720    60    | ~409.52MiB 2779k m3u8  | vp09.00.40.08 2779k video only          IOS
302     webm  1280x720    60    |  226.21MiB 1535k https | vp09.00.40.08 1535k video only          720p60, IOS, webm_dash
312     mp4   1920x1080   60    | ~755.08MiB 5125k m3u8  | avc1.64002A   5125k video only          IOS
299     mp4   1920x1080   60    |  504.90MiB 3427k https | avc1.64002A   3427k video only          1080p60, IOS, mp4_dash
617     mp4   1920x1080   60    | ~664.76MiB 4512k m3u8  | vp09.00.41.08 4512k video only          IOS
303     webm  1920x1080   60    |  397.18MiB 2696k https | vp09.00.41.08 2696k video only          1080p60, IOS, webm_dash
```

**Comment by bashonly** [comment]
Youtube has phased out formats 17 and 22, so it would make sense that the last legacy format (18) is being phased out as well

**Comment by bashonly** [comment]
btw OP: the solution is to install ffmpeg

**Comment by Trit34** [comment]
Reposting from #11164: 

I recently noticed that format 136 (720p30 MP4, useful when using a computer old enough – 2009, Core 2 Duo inside – that does not like 720p60) is missing sometimes, but becomes available later (**EDIT:** or after retrying). I don’t know if it’s like this for a long time or if it’s new.

**Comment by HajoBB** [comment]
Hello guys. I'm also having this fmt=18 issue

Requested format is not available. Use --list-formats for a list of available formats

But same like Trit34 said, even fmt=18 sometimes DOES play, I mean, I try to load a video with fmt=18, it fails, I try again just some seconds later, it works.

So what the heck is going on on YouTube? :D

What should I use as replacement for fmt=18 ??

**Comment by seproDev** [comment]
The alternative is having ffmpeg installed and letting yt-dlp merge the separate audio and video tracks.

**Comment by HajoBB** [comment]
Thanks for your reply. Well, we do this already for the download of videos. But for intstant playback, this is bad, no? And which formats to choose then .... is there a command like -f worst for that?

**Comment by seproDev** [comment]
You could use `-f wv+wa` to download the worst video and worst audio, but I think I would instead recommend using something like `-S "+size"` with no `-f` argument to download the smallest size file. If you also want to prefer avc video and aac audio, you could use `-S "vcodec:avc,acodec:aac,+size"` for example. Have a look at the documentation: https://github.com/yt-dlp/yt-dlp#sorting-formats

**Comment by HajoBB** [comment]
Thanks :)

**Comment by gamer191** [comment]
From a super quick test, it looks like format 18 is only being phased out on the creator and IOS clients

**Comment by HajoBB** [comment]
What do you mean by creator clients? But if we use yt-dlp normally on a linux system and it DOES what it does, sometimes offering format 18 and sometimes not, on the same video, then it seems rather globally than exceptionally?

**Comment by anikialt** [comment]
i am added:

--downloader ffmpeg -f 18

its works, thanks

**Comment by gamer191** [comment]
> then it seems rather globally than exceptionally?

The creator and iOS clients are the default clients. If you require format 18, use `--extractor-args "youtube:player-client=default,web_safari"`

(most other clients will also work)

> i am added:
> 
> --downloader ffmpeg -f 18
> 
> its works, thanks

I think it's random whether it works. `--downloader ffmpeg` shouldn't make any difference to the format extraction as far as I know

**Comment by rvisalm** [comment]
Commenting in case my experience helps anybody. I had '-f mp4' (with ffmpeg installed) and it was pulling video only formats since they've extinguished format 18. 

The solution that works best for me is to replace the format string by:  
' -f "bestvideo[height<=720][ext=mp4]+bestaudio[ext=m4a]" ' (it gets the best 720p quality video in mp4 and best audio, and then combines them)

**Comment by HajoBB** [comment]
For downloads this is all no problem - but if you use yt-dlp to be able to playback a video in your html5 player (instead of the native YouTube Player), then muxing formats is a big issue because yt-dlp first needs to download and mux before it can playback the muxed mp4 (or whatever). So that won't work anymore if the last combined format is phased out, right?

**Comment by DTrombett** [comment]
If you need to use it in a website you can use two different elements for audio and video and then sync it, like YouTube does

**Comment by HajoBB** [comment]
Thanks. Well this will become too complicated for me.

What about the solution of gamer191 - will that work? So they will not completely remove format 18 for the moment? But I think even this will just work for a while.

Anyway I get

WARNING: [youtube] No PO Token provided for web_safari client, which is required for working web_safari formats. You can manually pass a PO Token for this client with --extractor-args "youtube:po_token=web_safari+XXX"

**Comment by HajoBB** [comment]
I have tested a lot now, 

--extractor-args "youtube:player-client=default,web_safari"

seems to make the difference indeed, I always get format 18 - while without it's randomly working and not working.

But I still don't like that warning.

**Comment by HajoBB** [comment]
Manual says, you could be blocked using --extractor-args without a PO-Token.

So better follow this https://github.com/yt-dlp/yt-dlp/wiki/Extractors#other-methods-of-acquiring-a-po-token

and make your params look like this where you replace PO_TOKEN with your real token from the step before. At least I get no warnings now.

yt-dlp https://www.youtube.com/watch?v=X4RDB-mMDJg --list-formats --extractor-args "youtube:player-client=default;po_token=web_safari+PO_TOKEN"

**Comment by HajoBB** [comment]
1. I can confirm that --extractor-args with web_safari works.

2. You should enhance it, get a PO_Token and get rid of warnings which could otherwise block you.

Follow this tutorial to get the token: https://github.com/yt-dlp/yt-dlp/wiki/Extractors

and then use it like this:

yt-dlp https://www.youtube.com/watch?v=X4RDB-mMDJg --list-formats --extractor-args "youtube:player-client=default,web_safari;po_token=web_safari+YOURTOKEN"

**Comment by anikialt** [comment]
On other yt channel --extractor-args "youtube:player-client=default,web_safari" - is work

' -f "bestvideo[height<=720][ext=mp4]+bestaudio[ext=m4a]" '  - dont work (download audio separately)

Its became too hard operation - just downloads youtube video, need be a hucker

Will this problem be fixed in new versions yt-dlp?

**Comment by DTrombett** [comment]
If you have ffmpeg `yt-dlp URL` remains the easiest yet effective way to download a video. I'm not sure what you need precisely; a MP4? A premerged format?

**Comment by gamer191** [comment]
> From a super quick test, it looks like format 18 is only being phased out on the creator and IOS clients

False alarm. Youtube are NOT phasing out format 18, they are just rolling out a restriction that prevents web_creator working when logged out (you can verify this by using ``--extractor-arg "youtube:player_client=web_creator"``, which will sometimes give `ERROR: [youtube] VIDEOID: Please sign in`)
Thanks @seproDev for reporting that issue, and @bashonly for connecting the dots

As for the iOS client, it's never had format 18, according to https://github.com/zerodytrash/YouTube-Internal-Clients/
> WARNING: [youtube] No PO Token provided for web_safari client, which is required for working web_safari formats. You can manually pass a PO Token for this client with --extractor-args "youtube:po_token=web_safari+XXX"

I believe that only applies for formats that aren't pre-merged. In my opinion it should probably be re-worded
> `' -f "bestvideo[height<=720][ext=mp4]+bestaudio[ext=m4a]" '` - dont work (download audio separately)

That command is supposed to download video and audio separately. I think @rvisalm incorrectly assumed that the main usecase of format 18 is to limit video quality to 720p
> Will this problem be fixed in new versions yt-dlp?

Format 18 will be fixed by #11190. The argument will always be needed for the higher quality premerged* formats (91-96, possibly higher 2 digit numbers).
*unlike format 18, formats 91-96 are delivered over m3u8, meaning they are delivered in chunks and merged by yt-dlp. So they won't be as useful as format 18, but they will be higher quality
> I'm not sure what you need precisely; a MP4? A premerged format?

Most likely the latter

**Comment by gamer191** [comment]
Looks like it was actually discovered by @arsh1219 (who posted it at https://github.com/yt-dlp/yt-dlp/issues/11185)

So I'll close this as a duplicate of #11185

**Comment by Trit34** [comment]
> For downloads this is all no problem - but if you use yt-dlp to be able to playback a video in your html5 player (instead of the native YouTube Player), then muxing formats is a big issue because yt-dlp first needs to download and mux before it can playback the muxed mp4 (or whatever). So that won't work anymore if the last combined format is phased out, right?

`mpv --ytdl-format=136+140 [URL]` works fine, but it is quite throttled (between 50 and 400 kiB/s, depending on the encoding parameters) and I recommend to wait some minutes in pause mode before you can play it without buffering. There is no throttling if you first download it before playing from your computer.

**Comment by HalvorRaknes** [comment]
I created https://github.com/yt-dlp/yt-dlp/issues/11164 which was merged into present issue a couple of weeks ago.

I contend that `-f worst` should be a failsafe option, and if devs aren't prepared to maintain it as such it should be removed, OR AT THE VERY LEAST, it should be removed, leaving only `wv` and `wa`.

And by 'failsafe' I don't mean being contingent on there being a merged format (that includes both video and audio).

Leaving this option as it is at present will cause unnecessary frustration to unsavvy users.

**Comment by Grub4K** [comment]
> I contend that `-f worst` should be a failsafe option, and if devs aren't prepared to maintain it as such it should be removed leaving only `wv` and `wa`.

You are misrepresenting the problem. `-f` is explicit and has to fail if a format is not available. It is on the user to pass a format selector that does not fail. `worst` has valid usecases for sites other than YouTube, or when you want it to error if no premerged format is available, ... It should definately NOT be removed.

In your case I would assume you want something like `-f "worst/wv*+wa"`.

**Comment by seproDev** [comment]
To further expand on that, I totally agree that `worst` and `bad` are unintuitive. yt-dlp even warns when you use `-f best` due to the easy confusion the name causes.
This is however not something that can be easily change. Thousands of scripts and other tools depend on the current behavior. Yes, if you were re-designing this you would go a different way, but given the enormous friction a change would cause this is unlikely to be ever changed.

**Comment by HalvorRaknes** [comment]
Then a warning in all caps should appear to any user how requests a `-f worst` download.

**Comment by error-reporting** [comment]
> > For downloads this is all no problem - but if you use yt-dlp to be able to playback a video in your html5 player (instead of the native YouTube Player), then muxing formats is a big issue because yt-dlp first needs to download and mux before it can playback the muxed mp4 (or whatever). So that won't work anymore if the last combined format is phased out, right?
> 
> `mpv --ytdl-format=136+140 [URL]` works fine, but it is quite throttled (between 50 and 400 kiB/s, depending on the encoding parameters) and I recommend to wait some minutes in pause mode before you can play it without buffering. There is no throttling if you first download it before playing from your computer.
add --extractor-args "youtube:player_client=web_safari"

**Comment by error-reporting** [comment]
> btw OP: the solution is to install ffmpeg

OP: the solution is to add --extractor-args "youtube:player_client=web_safari"
